pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -76,6 +87,14 @@ def configure_checkpoint_callbacks(self, checkpoint_callback: Union[ModelCheckpo
         if not self._trainer_has_checkpoint_callbacks() and checkpoint_callback is True:
             self.trainer.callbacks.append(ModelCheckpoint(dirpath=None, filename=None, mode='min'))
 
+    def configure_swa_callbacks(self):
+        for cb in self.trainer.callbacks:
+            if isinstance(cb, StochasticWeightAveraging):
+                return
```

Comment:
I think a `MisconfigurationException` is in place here (similar to progressbar) since multiple of this callback does not make sense?

Reply:
I don't there is a need a `MisconfigurationException`. It would be wierd to break if the user provides a StochasticWeightAveraging and `_stochastic_weight_avg` is True

Reply:
maybe a warning, that the flag has no effect then?

Reply:
I think we should do the same logic as in `configure_checkpoint_callbacks`

```python
if has_swa_callback and not use_swa:
    raise Misconfig

if not has_swa_callback and use_swa:
     set_default_swa
```